internal/deviations/deviations.go

```diff
@@ -65,27 +65,91 @@ package deviations
 
 import (
 	"flag"
+	"fmt"
+	"regexp"
 
 	log "github.com/golang/glog"
 	"github.com/openconfig/featureprofiles/internal/metadata"
 	mpb "github.com/openconfig/featureprofiles/proto/metadata_go_proto"
 	"github.com/openconfig/ondatra"
 )
 
-// lookupDutDeviations returns the deviations for the specified dut.
-func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
+func matchRegex(regex string, deviceInfo string) bool {
+	match, err := regexp.MatchString(regex, deviceInfo)
+	if err != nil {
+		log.Errorf("Error with regex match %v", err)
```

Comment:
Agree, if the regex is invalid it should be a hard error. Logged with error with Exitf instead of Errorf here so that this function returns a bool and keeps the if condition below simple.

Reply:
Ug, Go code should really handle and propagate errors, leaving the "must" behavior at the top-level.
https://google.github.io/styleguide/go/decisions.html#must-functions

I know you moved the matchRegex calls into the if-condition at my suggestion, but at the time I didn't connect that suggestion to my suggestion here that this should result in a hard error. Idiomatic Go would just prefer that you propagate the error here and nest the matchRegex call. In fact, you can just get rid of matchRegex at his point and inline the regexp.MatchString call directly.

Reply:
done.